setup/building-stratum-deps.md

```diff
@@ -0,0 +1,183 @@
+# Building the Stratum Dependencies
+
+Stratum is the component of `infrap4d` that implements the P4Runtime and gNMI
+(OpenConfig) services. It requires that a number of third-party libraries
+be built and installed on the system.
+
+This document explains how to build and install the Stratum dependencies.
+
+> Note: For the Intel IPU E2100, see
+[Building Stratum Dependencies for ES2K](building-stratum-deps-es2k.md).
+
+## 1 Prerequisites
+
+### 1.1 CMake
+
+`cmake 3.15` or above must be installed.
+
+Avoid versions 3.24.x and 3.25.x. They have an issue that causes the Protobuf
+build to fail.
+
+### 1.2 OpenSSL
+
+`OpenSSL 1.1` must be installed. The Stratum dependencies use OpenSSL instead
+of building and installing BoringSSL.
+
+## 2 Source Code
+
+There are two ways to obtain the source code: from the open-source repository
+on GitHub, or as a source package.
+
+### 2.1 GitHub Repository
+
+The Stratum dependencies can be built using a script that is part of the
+IPDK networking-recipe. To clone the repository:
+
+```bash
+git clone --recursive https://github.com/ipdk-io/networking-recipe.git ipdk.recipe
+```
+
+You may omit the `--recursive` option if you are only interested in building
+the dependencies.
+
+You may substitute your own local directory name for `ipdk.recipe`.
+
+The build script for the stratum dependencies is in the `setup` directory.
+
+### 2.2 Source Package
+
+You may receive the source code for the Stratum dependencies in the form of
+an RPM file, a Debian file, or a tarball.
+
+If this is case, use the appropriate command to unpack the source files.
+
+## 3 Install Location
+
+You will need to decide where on your system you would like to install the
+dependencies. This location (the "install prefix") must be specified when you
+configure the build.
+
+It is recommended that you _not_ install the dependency libraries in `/usr` or
+`/usr/local`. It is easier to upgrade or do a clean build if packages that are
+updated on different cadences are kept separate from one another.
+
+The `CMAKE_INSTALL_PREFIX` option is used to specify the directory in which
+the dependencies should be installed.
+
+If you plan to install the dependency libraries in a system directory, you will
+need to log in as `root` or run from an account that has `sudo` privilege.
+
+## 4 Downloaded Source vs. Source Package
+
+By default, the cmake build script downloads the source code for each module
+and then builds and installs it.
+
+If you are working from a source distribution, or if the source files have
+were downloaded by a previous build, you can override the download stage by
+specifying:
+
+```cmake
+  -DDOWNLOAD=no -DFORCE_PATCH=yes
+```  
+
+when you configure the cmake build.
+
+## 5 Build Options
+
+The CMake build script supports the following configuration options.
+
+| Option | Type | Description |
```

Comment:
The table looks weird in GitHub review. please verify that when rendered, it looks ok

Reply:
It looks pretty good. To see for yourself, click the ellipsis (. . .) in the upper right corner of the file panel (to the right of the "Comment on this" icon, below the "Review Changes" button) and select "View File". GitHub will give you preview of the page.